Makefile

```diff
@@ -41,42 +41,35 @@ include maint-lib/makelib.mk
 # All flavor options that can be passed to FLAVORS
 # CEPH_VERSION,ARCH,OS_NAME,OS_VERSION,BASEOS_REG,BASEOS_REPO,BASEOS_TAG
 ALL_BUILDABLE_FLAVORS := \
-	luminous,x86_64,ubuntu,16.04,_,ubuntu,16.04 \
-	jewel,x86_64,ubuntu,16.04,_,ubuntu,16.04 \
-	jewel,x86_64,ubuntu,14.04,_,ubuntu,14.04 \
-	kraken,x86_64,ubuntu,16.04,_,ubuntu,16.04 \
-	luminous,x86_64,centos,7,_,centos,7 \
-	jewel,x86_64,centos,7,_,centos,7 \
-	kraken,x86_64,centos,7,_,centos,7 \
-	luminous,x86_64,opensuse,42.3,_,opensuse,42.3 \
+	luminous,ubuntu,16.04 \
+	jewel,ubuntu,16.04 \
+	jewel,ubuntu,14.04 \
+	kraken,ubuntu,16.04 \
+	luminous,centos,7 \
+	jewel,centos,7 \
+	kraken,centos,7 \
+	luminous,opensuse,42.3 \
 
 # ==============================================================================
 # Build targets
 .PHONY: all stage build build.parallel build.all push
 
 stage.%:
-	@$(call set_env_var,CEPH_VERSION,$*) $(call set_env_var,ARCH,$*) \
-	$(call set_env_var,OS_NAME,$*) $(call set_env_var,OS_VERSION,$*) \
-	$(call set_env_var,BASEOS_REG,$*) $(call set_env_var,BASEOS_REPO,$*) \
-	$(call set_env_var,BASEOS_TAG,$*) $(call set_env_var,IMAGES_TO_BUILD,$*) \
-	$(call set_env_var,STAGING_DIR,$*) $(call set_env_var,BASE_IMAGE,$*) \
-	$(call set_env_var,DAEMON_BASE_IMAGE,$*) $(call set_env_var,DAEMON_IMAGE,$*) \
-	$(call set_env_var,RELEASE,$*) \
-	sh -c maint-lib/stage.py
+	@$(call set_env_vars,$*) sh -c maint-lib/stage.py
 
 daemon-base.%: stage.%
-	@$(call set_env_var,STAGING_DIR,$*) ; $(MAKE) -C $$STAGING_DIR/daemon-base $(MAKECMDGOALS) \
-	$(call set_env_var,DAEMON_BASE_IMAGE,$*)
+	@$(call set_env_vars,$*); $(MAKE) -C $$STAGING_DIR/daemon-base \
+	  $(call set_env_vars,$*) $(MAKECMDGOALS)
 
 daemon.%: daemon-base.%
-	@$(call set_env_var,STAGING_DIR,$*) ; $(MAKE) -C $$STAGING_DIR/daemon $(MAKECMDGOALS) \
-	$(call set_env_var,DAEMON_IMAGE,$*)
+	@$(call set_env_vars,$*); $(MAKE) $(call set_env_vars,$*) -C $$STAGING_DIR/daemon \
+	  $(call set_env_vars,$*) $(MAKECMDGOALS)
 
 do.image.%: daemon.% ;
 
-stage: $(foreach p, $(FLAVORS), stage.$(p)) ;
-build: $(foreach p, $(FLAVORS), do.image.$(p)) ;
-push:  $(foreach p, $(FLAVORS), do.image.$(p)) ;
+stage: $(foreach p, $(FLAVORS), stage.$(HOST_ARCH),$(p)) ;
```

Comment:
So this implicit HOST_ARCH becomes ARCH ? Couldn't we have the same name across the code to avoid confusions ?